### PR TITLE
Merchant endpoints

### PR DESCRIPTION
**Pull request by zpalmquist**
Hey Zach! So, I wanted you to tell me if I should be doing it like I am and having separate controllers for each of these methods, as the routes are pointing to different controllers and the one action in each controller

Would love feedback on 
˚ Routes
˚ RESTfulness
˚ If I need to pass arguments to the class methods
˚ Any other "standards" stuff that could make my code better.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -6,6 +6,8 @@
         get 'find' => "searches#show"
         get 'find_all' => "searches#index"
         get 'random' => "searches#random"
+        get '/:id/items' => "item_searches#merchant_items"
+        get '/:id/invoices' => "invoice_searches#merchant_invoices"
```
Comment (zacharylandes) [inline config/routes.rb]: try to keep actions down to only show or index, and alter controller names as needed to gain clarification into what its doing. I think the controller names are good in these cases

File: app/models/merchant.rb [hunk 2]
```diff
@@ -4,4 +4,16 @@ class Merchant < ApplicationRecord
   has_many :customers, through: :invoices
   has_many :invoice_items, through: :invoices
   has_many :transactions, through: :invoices
+
+  def self.all_items(id)
```
Comment (zacharylandes) [inline app/models/merchant.rb]: I don't think you should need to pass ids to class methods. Self should be good enough...pretty sure

File: config/routes.rb (hunk 1 above)
Comment (zacharylandes) [reply config/routes.rb]: its been a sec but I think restful protocol would be /merchants/:id/items and so forth

Comment (freshouttheovenprogrammes) [reply config/routes.rb]: I see what you mean, so something like

`get '/:id/items' => "item_searches#index
  get '/:id/invoices' => "invoice_searches#index" `

**Pull request closed without merging**